openerp/osv/fields.py

```diff
@@ -1078,7 +1081,10 @@ def unlink_all():
             elif act[0] == 5:
                 unlink_all()
             elif act[0] == 6:
-                unlink_all()
+                keep_ids = set(act[2])
+                for act1 in [v for v in values if v[0] in (1, 4)]:
+                    keep_ids.add(act1[1])
+                unlink_all(keep_ids)
```

Comment:
`unlink_all(keep_ids)` is quite a confusing call..  we might want to rename stuff here ;-)